src/Raven.Server/Commercial/LetsEncryptClient.cs

```diff
@@ -761,9 +783,15 @@ public JwsMessage Encode<TPayload>(TPayload payload, JwsHeader protectedHeader)
                     protectedHeader.Key = _jwk;
                 }
 
+                // From: https://tools.ietf.org/html/rfc8555#section-7.5.1
+                var encodedPayload = "{}".Equals(payload)
+                    ? Base64UrlEncoded("{}")
+                    : Base64UrlEncoded(JsonConvert.SerializeObject(payload));
+
                 var message = new JwsMessage
                 {
-                    Payload = Base64UrlEncoded(JsonConvert.SerializeObject(payload)),
+				    // From: https://tools.ietf.org/html/rfc8555#section-6.3
+                    Payload =  string.Empty.Equals(payload) ? string.Empty : encodedPayload,
```

Comment:
Put this logic with the encodedPayload, so we'll have a `case` for the various options